### ISSUE TITLE
libcurl-gnutls error on Fedora

### ISSUE THREAD

**Issue by ziggy42** [shiftkey/desktop#21]
### Description

I'm not able to clone a repository with the app installed via rpm:
```
/opt/GitHubDesktop/resources/app/git/libexec/git-core/git-remote-https: error while loading shared libraries: libcurl-gnutls.so.4: cannot open shared object file: No such file or directory
```

### Version

**GitHub Desktop version:** 1.0.12-linux1

**OS version:** Fedora 27

### Steps to Reproduce

1. Try to clone something

**Comment by shiftkey** [comment]
This is a known issue with `dugite-native` as it's dynamically linked to `libcurl-gnutls`, which might not be present on the machine: https://github.com/desktop/dugite-native/issues/42

I've been exploring a bit about how to statically link Git so that it doesn't depend on these libraries, but that's far from being ready to use. I also am curious about the `electron-builder` [option](https://www.electron.build/configuration/linux) to provide additional dependencies when building the installer:

> `depends` `Array<String>` - Package dependencies. Defaults to `["gconf2", "gconf-service", "libnotify4", "libappindicator1", "libxtst6", "libnss3"]`.

For now the best advice is to install this missing dependency to workaround the issue.

**Comment by Croydon** [comment]
Has somebody found a package for Fedora 27 which has exactly this library?

**Comment by shiftkey** [comment]
This seems to workaround the issue for me on my FC27 VM:

```shellsession
$ cd /usr/lib64/
$ ls -l libcurl*
lrwxrwxrwx. 1 root root     16 Nov 30 09:25 libcurl.so -> libcurl.so.4.4.0
lrwxrwxrwx. 1 root root     16 Nov 30 09:25 libcurl.so.4 -> libcurl.so.4.4.0
-rwxr-xr-x. 1 root root 532528 Nov 30 09:25 libcurl.so.4.4.0
$ sudo ln -s libcurl.so.4 libcurl-gnutls.so.4
[sudo] password for parallels: 
$ ls -l libcurl*
lrwxrwxrwx. 1 root root     12 Jan 15 10:33 libcurl-gnutls.so.4 -> libcurl.so.4
lrwxrwxrwx. 1 root root     16 Nov 30 09:25 libcurl.so -> libcurl.so.4.4.0
lrwxrwxrwx. 1 root root     16 Nov 30 09:25 libcurl.so.4 -> libcurl.so.4.4.0
-rwxr-xr-x. 1 root root 532528 Nov 30 09:25 libcurl.so.4.4.0
```

**Comment by aia832003** [comment]
Thanks for this @shiftkey this was my roadblock this morning for starting to experiment with github.

**Comment by Bodyash** [comment]
@shiftkey Nice, you make it work on Fedora 27!

**Comment by shiftkey** [comment]
Just tested out https://github.com/desktop/desktop/pull/5388 on CentOS and confirmed it no longer needs `libcurl-gnutls`, so hopefully `v1.3.4` will resolve this issue without needing to symlink any more.

**Comment by ct-martin** [comment]
Adding that this bug exists on Fedora 28 using GitHub Desktop 1.3.4 RC and that the workaround works as well

**Comment by shiftkey** [comment]
@ct-martin you're going to have to wait for a future update because https://github.com/desktop/desktop/pull/5388 hasn't been merged yet

**Comment by shiftkey** [comment]
This should be fixed in 1.4.0 because of https://github.com/desktop/desktop/pull/5478 but I'm trying to figure out whether I got the dependencies right for Ubuntu over in #60. Anyway, please submit follow-up issues if you spot something and we can dig into them.
